src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_INITIAL_CAPACITY = 16;
+    public static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int currentCapacity;
+    private int currentThreshold;
+    private Node<K,V>[] table;
+    private int size;
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        currentThreshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (++size > currentThreshold) {
```

Comment:
```suggestion
        if (size == currentThreshold) {
```
let's change `size` only where it's needed